codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -254,6 +260,60 @@ protected void encode(ChannelHandlerContext ctx, ByteBuf in, ByteBuf out) throws
         }
     }
 
+    @Override
+    protected ChannelPromise encode(ChannelHandlerContext ctx, ByteBuf msg, ByteBuf out, ChannelPromise promise)
+            throws Exception {
+        try {
+            encode(ctx, msg, out);
+        } catch (Exception e) {
+            if (bufferPromise != null) {
+                bufferPromise.tryFailure(e);
```

Comment:
actually I think we should use `setFailure`

Reply:
++ done